pkg/provider/resource_project.go

```diff
@@ -80,73 +96,136 @@ func (p *ProjectResource) Schema() *schema.Resource {
 	}
 }
 
-func (p *ProjectResource) Create(ctx context.Context, d *schema.ResourceData, meta any) diag.Diagnostics {
-	client := api.BuildAPI(meta).ProjectClient()
-	project_name := d.Get("project_name").(string)
+// Configure adds the provider configured client to the data source.
+func (r *projectResource) Configure(_ context.Context, req resource.ConfigureRequest, _ *resource.ConfigureResponse) {
+	if req.ProviderData == nil {
+		return
+	}
+
+	r.client = req.ProviderData.(*api.API)
+}
+
+type cloudProvider struct {
+	CloudProviderId   string `tfsdk:"cloud_provider_id"`
+	CloudProviderName string `tfsdk:"cloud_provider_name"`
+}
+
+type Project struct {
+	ID             types.String    `tfsdk:"id"`
+	ProjectID      types.String    `tfsdk:"project_id"`
+	ProjectName    types.String    `tfsdk:"project_name"`
+	UserCount      types.Int64     `tfsdk:"user_count"`
+	ClusterCount   types.Int64     `tfsdk:"cluster_count"`
+	CloudProviders []cloudProvider `tfsdk:"cloud_providers"`
+}
 
-	projectId, err := client.Create(ctx, project_name)
+// Create creates the resource and sets the initial Terraform state.
+func (p projectResource) Create(ctx context.Context, req resource.CreateRequest, resp *resource.CreateResponse) {
+	var config Project
+	diags := req.Config.Get(ctx, &config)
+	resp.Diagnostics.Append(diags...)
+	if resp.Diagnostics.HasError() {
+		return
+	}
 
+	projectId, err := p.client.ProjectClient().Create(ctx, config.ProjectName.ValueString())
 	if err != nil {
-		return fromBigAnimalErr(err)
+		resp.Diagnostics.AddError("Error creating project", "Could not create project, unexpected error: "+err.Error())
+		return
 	}
 
-	d.SetId(projectId)
+	project, err := p.client.ProjectClient().Read(ctx, projectId)
```

Comment:
As a general idea, can we move this `p.client.ProjectClient()` to the Configure function, where we set the provider configured client? 🤔 So that, we can call the client functions as:
```
p.client.Create
```
It looks more readable to me, rather than `p.client.ProjectClient().Create`. 
We shouldn't initialize a ProjectClient every time we call a method, IMHO.

Reply:
that's great, follow your thought